spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/MockFactory.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2002-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.ip.tcp.connection;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Gengwu Zhao
+ *
+ * @since 6.3.1
+ *
+ */
+
+public class MockFactory {
```

Comment:
The name for the class is too generic to be as a public API.
Plus it really does not reflect what is going on in the class.
The code style in the class does not follow our rules.
Please, use `check` Gradle command locally to catch all the code style problems before pushing.

I commented before that I prefer to not have a mixes between test classes since their logic might diverse in the future.
So, while the change you are doing is OK now and makes less code lines, it can become burden in the future to support such a generic code.

Therefore I'm OK with extracting common code into static methods of the same test class.
This will ensure that change into one test class won't effect others.

Plus it is better to be more specific with names.
This `MockFactory` class name is not about what it does.
The name of the `makeMockFactory()` method is also not fully about what it is doing.